dawn/src/dawn/Compiler/DawnCompiler.h

```diff
@@ -37,11 +37,22 @@ class DawnCompiler : NonCopyable {
   /// @brief Initialize the compiler by setting up diagnostics
   DawnCompiler(Options* options = nullptr);
 
-  /// @brief Compile the SIR using the provided code generation routine
-  /// @returns compiled TranslationUnit on success, `nullptr` otherwise
-  std::unique_ptr<codegen::TranslationUnit> compile(std::shared_ptr<SIR> const& SIR);
+  /// @brief Apply parallelizer, code optimization, and generate
+  std::unique_ptr<codegen::TranslationUnit> compile(std::shared_ptr<SIR> const& stencilIR);
```

Comment:
How would you like the API to look to pass and IIR file?

Reply:
I'd like to overload `compile` with an IIR. Deserialization from a file should not be a concern for `DawnCompiler`, as it breaks the S rule in https://en.wikipedia.org/wiki/SOLID

Reply:
Currently the driver code for dawn is `compile`. I would agree that the module that optimizes should not be the one that worries about serialization. But isn't the sole purpose of the `DawnCompiler` to drive dawn whereas the `Optimizer` should worry about optimisation?

Reply:
Yes, I agree that in the current framework and using incremental changes that it makes sense to leave that in the compile method 😄